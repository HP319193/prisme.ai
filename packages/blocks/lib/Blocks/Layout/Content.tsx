import { useCallback, useLayoutEffect, useMemo, useRef, useState } from 'react';
import { BlockContext, useBlock } from '../../Provider';
import { Content as IContent } from './context';
import tw from '../../tw';
import { useBlocks } from '../../Provider/blocksContext';

interface ContentProps {
  content: IContent;
  onUnmount: () => void;
  removed: boolean;
  className?: string;
}

export const Content = ({
  content: { blocks },
  className,
  onUnmount,
  removed,
}: ContentProps) => {
  const {
    utils: { BlockLoader },
  } = useBlocks();
  const [animationClassName, setAnimationClassName] = useState(
    tw`translate-x-full`
  );
  const containerEl = useRef<HTMLDivElement>(null);

  useLayoutEffect(() => {
    setTimeout(() => setAnimationClassName(''));
  }, []);

  useLayoutEffect(() => {
    if (!removed) return;
    setTimeout(() => setAnimationClassName(tw`translate-x-full`));
    setTimeout(onUnmount, 200);
  }, [removed]);

  return (
    <div
      ref={containerEl}
      className={tw`${className} content-stack__content content transition-transform  ${animationClassName}`}
    >
      {blocks &&
        blocks.map(({ block, url, onInit, ...config }, index) => (
          <div
            key={index}
            className={tw`flex content__block-container block-container snap-start`}
          >
            <BlockLoader
              config={config}
              name={block}
              container={containerEl.current || undefined}
            />
          </div>
        ))}
    </div>
  );
};
export default Content;
