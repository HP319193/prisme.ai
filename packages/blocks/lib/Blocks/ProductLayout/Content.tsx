import ContentPanel from './ContentPanel';
import { isBlock, isRenderProp } from '../utils/getContentType';
import { ContentProps, ProductLayoutProps } from './types';
import GenericBlock from '../utils/GenericBlock';

function isContentProps(content: any): content is ContentProps {
  if (!content) return false;
  const { title, description, tabs } = content as ContentProps;
  return !!(title || description || tabs);
}

export const Content = ({
  content,
}: {
  content: ProductLayoutProps['content'];
}) => {
  if (isRenderProp(content)) return content;
  if (isBlock(content))
    return (
      <GenericBlock content={content} className="product-layout-content" />
    );
  if (!isContentProps(content)) return null;
  return (
    <div className="product-layout-content">
      {content.title && (
        <GenericBlock
          content={content.title}
          className="product-layout-content-title"
          ifString={({ content, className }) => (
            <div className={className}>{content}</div>
          )}
        />
      )}
      {content.description && (
        <GenericBlock
          content={content.description}
          className="product-layout-content-description"
          ifString={({ content, className }) => (
            <div className={className}>{content}</div>
          )}
        />
      )}
      <ContentPanel {...content} />
    </div>
  );
};

export default Content;
