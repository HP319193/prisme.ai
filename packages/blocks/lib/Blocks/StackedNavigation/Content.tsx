import { useLayoutEffect, useRef, useState } from 'react';
import { Content as IContent } from './context';
import tw from '../../tw';
import { BlocksList } from '../BlocksList';

interface ContentProps {
  content: IContent;
  onUnmount: () => void;
  removed: boolean;
  className?: string;
}

export const Content = ({
  content: { blocks },
  className,
  onUnmount,
  removed,
}: ContentProps) => {
  const [animationClassName, setAnimationClassName] = useState(
    tw`translate-x-full`
  );
  const containerEl = useRef<HTMLDivElement>(null);

  useLayoutEffect(() => {
    setTimeout(() => setAnimationClassName(''));
  }, []);

  useLayoutEffect(() => {
    if (!removed) return;
    setTimeout(() => setAnimationClassName(tw`translate-x-full`));
    setTimeout(onUnmount, 200);
  }, [removed]);

  const legacyBlocks = blocks.map(({ block, slug, ...rest }) => ({
    slug: slug || block || '',
    ...rest,
  }));

  return (
    <div
      ref={containerEl}
      className={tw`${className} content-stack__content content transition-transform  ${animationClassName}`}
    >
      {blocks && (
        <BlocksList
          blocks={legacyBlocks}
          className={tw`flex content__block-container block-container snap-start`}
        />
      )}
    </div>
  );
};
export default Content;
