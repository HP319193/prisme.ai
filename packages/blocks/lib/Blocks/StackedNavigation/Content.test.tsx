import renderer, { act } from 'react-test-renderer';
import Content from './Content';

jest.mock('../../Provider/blocksContext', () => {
  const mock = {
    utils: {
      BlockLoader: jest.fn(function BlockLoader({ onLoad }: any) {
        onLoad && onLoad();
        return null;
      }),
    },
  };
  const useBlocks = () => mock;
  return {
    useBlocks,
  };
});

it('should render with only a title', () => {
  jest.useFakeTimers();
  const content: Parameters<typeof Content>[0]['content'] = {
    title: 'Foo',
    blocks: [],
  };
  const onUnmount = jest.fn();
  const root = renderer.create(
    <Content content={content} onUnmount={onUnmount} removed={false} />
  );
  expect(root.toJSON()).toMatchSnapshot();

  act(() => {
    jest.runAllTimers();
  });

  expect(root.toJSON()).toMatchSnapshot();
});

it('should be removed', () => {
  jest.useFakeTimers();
  const content: Parameters<typeof Content>[0]['content'] = {
    title: 'Foo',
    blocks: [],
  };
  const onUnmount = jest.fn();
  const root = renderer.create(
    <Content content={content} onUnmount={onUnmount} removed />
  );
  expect(root.toJSON()).toMatchSnapshot();

  act(() => {
    jest.runAllTimers();
  });

  expect(root.toJSON()).toMatchSnapshot();
});

it('should render with blocks', () => {
  const content: Parameters<typeof Content>[0]['content'] = {
    title: 'Foo',
    blocks: [
      {
        block: 'RichText',
        content: 'Hello World',
        onInit: 'initFoo',
      },
    ],
  };
  const onUnmount = jest.fn();

  const root = renderer.create(
    <Content content={content} onUnmount={onUnmount} removed={false} />
  );

  expect(root.toJSON()).toMatchSnapshot();
});
