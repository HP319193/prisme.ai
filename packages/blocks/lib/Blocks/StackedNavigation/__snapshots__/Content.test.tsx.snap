// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be removed 1`] = `
<div
  className="flex content-stack__content content transition-transform translate-x-full"
>
  <div
    className="pr-block-blocks-list flex flex-1 flex-col max-w-full overflow-auto"
  />
</div>
`;

exports[`should be removed 2`] = `
<div
  className="flex content-stack__content content transition-transform translate-x-full"
>
  <div
    className="pr-block-blocks-list flex flex-1 flex-col max-w-full overflow-auto"
  />
</div>
`;

exports[`should render with blocks 1`] = `
<div
  className="flex content-stack__content content transition-transform translate-x-full"
>
  <div
    className="pr-block-blocks-list flex flex-1 flex-col max-w-full overflow-auto"
  >
    <div
      className="pr-block-blocks-list__block pr-block-blocks-list__block--0 flex flex-1 content__block-container block-container snap-start max-h-full"
    />
  </div>
</div>
`;

exports[`should render with only a title 1`] = `
<div
  className="flex content-stack__content content transition-transform translate-x-full"
>
  <div
    className="pr-block-blocks-list flex flex-1 flex-col max-w-full overflow-auto"
  />
</div>
`;

exports[`should render with only a title 2`] = `
<div
  className="flex content-stack__content content transition-transform"
>
  <div
    className="pr-block-blocks-list flex flex-1 flex-col max-w-full overflow-auto"
  />
</div>
`;
